vtex-apps/phone-field
# Add PhoneField input component implementation

#### What problem is this solving?
This PR adds the initial version of the `PhoneField` component, alongside some default rules to get started.

##### _Note to reviewers:_

This is a pretty large pull request in number of files, but most of them (precisely 209) are SVG files for the flags icons, which could be extracted later to another app, so don't be scared by the total number of lines changed.

Another thing that I should mention is that this component uses a new component named `Listbox`. I used this component (from the `reach-ui` third-party library) because of a couple of things:
- Styleguide's `Dropdown` component doesn't meet my needs for this component, since it's just a simple HTML `<select>` and we can't customize it to have the flag icons inside the component;
- This component, and the whole library of `reach-ui` components, are accessible by default, which is a great plus for us because it solves a lot of accessibility problems that we would have by building this component from scratch. Eventually we may migrate this component to Styleguide.

EDIT: if you want to "skip" all the SVG files during review (all files inside `react/icons/`), you can use the following snippet to toggle the "reviewed" button on GitHub UI. All you need to do is to be in the the [files](https://github.com/vtex-apps/phone-field/pull/1/files) tab and run the script on the console.

⚠️ This may slow down your browser/tab temporarily.

```javascript
Array.from(document.querySelectorAll('.file'))
  .filter(node => node.querySelector('[title^="react/icons/"]'))
  .forEach(node => node.querySelector('.js-reviewed-toggle').click())
```

#### How should this be manually tested?
You can access my changes in [this workspace](https://chkio--checkoutio.myvtex.com/). To see this component you need to do a few steps first.

1. Clear your `checkout.vtex.com` cookie and delete the `orderform` entry from `localStorage`.
2. Add any item to your cart, [this one will do](https://chkio--checkoutio.myvtex.com/teste-so-delivery-copy-256--copy-257--copy-261--copy-262--copy-263--copy-264-/p).
3. Open the minicart and press the "Go to Checkout" button, then in the cart page click "Checkout".
4. Click on the edit button next to the "Profile" label.

Now you should see the phone field component in all of it's glory.

#### Checklist/Reminders


#### Screenshots or example usage
<img width="401" alt="image" src="https://user-images.githubusercontent.com/10223856/76994089-a838e580-692c-11ea-8e89-ea4fcc7a51c6.png">
<img width="405" alt="image" src="https://user-images.githubusercontent.com/10223856/76995145-306bba80-692e-11ea-9f3b-847627dfc49f.png">

#### Type of changes


✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements
